services/userGroup.service.ts

```diff
@@ -0,0 +1,79 @@
+import { UserGroup, UserGroupModel } from "../models/userGroup.model"
+import { taskSchema } from "../models/task.model"
+import { v4 as uuidv4 } from "uuid"
+import ErrorHandler from "../errors/errorHandler"
+import mongoose from "mongoose"
+
+const TaskModel = mongoose.model("Task", taskSchema)
+
+export default class UserGroupService {
+  getAllUserGroups() {
+    return UserGroupModel.find(
+      {},
+      {
+        _id: 0,
+        groupId: 1,
+        groupName: 1,
+        groupDescription: 1,
+        groupMembers: 1,
+      }
+    )
+  }
+  async createUserGroup(userGroupData: UserGroup) {
+    const id = uuidv4()
+    const creatingUserGroup = await UserGroupModel.create({
+      groupId: id,
+      groupName: userGroupData.groupName,
+      groupDescription: userGroupData.groupDescription,
+      groupMembers: userGroupData.groupMembers,
+    })
+    var createdGroupName = userGroupData.groupName
+    if (creatingUserGroup) {
+      return { createdGroupName }
+    } else {
+      throw new ErrorHandler()
+    }
+  }
+  async updateUserGroup(id: string, userGroupData: UserGroup) {
+    const findGroup = await UserGroupModel.findOne({ groupId: id })
+    const userGroup = {
+      groupName: userGroupData.groupName,
+      groupDescription: userGroupData.groupDescription,
+      groupMembers: findGroup?.groupMembers?.concat(userGroupData.groupMembers),
+    }
+    if (userGroupData.groupName == "") {
```

Comment:
these 2 if statements are weird, why are they necessary?

Reply:
If Admin decides to just add members to the group and not change group description or name or both, then the fields come to backend as empty strings, in order to not overwrite the valid groupName or Description these "IF" statements pull currently active names and replace them so that only group members get updated